Simple Arithmetic
You could use arithmetic operators + - * / directly between NumPy arrays, but this section discusses an extension of the same where we have functions that can take any array-like objects e.g. lists, tuples etc. and perform arithmetic conditionally
Số học đơn giản
Bạn có thể sử dụng trực tiếp các toán tử số học + - * / giữa các mảng NumPy, nhưng phần này thảo luận về phần mở rộng tương tự trong đó chúng ta có các hàm có thể nhận bất kỳ đối tượng giống mảng nào, ví dụ: danh sách, bộ dữ liệu, v.v. và thực hiện số học có điều kiện ..

Arithmetic Conditionally: means that we can define conditions where the arithmetic operation should happen.
Số học có điều kiện: có nghĩa là chúng ta có thể xác định các điều kiện mà phép toán số học sẽ xảy ra.

All of the discussed arithmetic functions take a where parameter in which we can specify that condition.
Tất cả các hàm số học được thảo luận đều có một tham số where trong đó chúng ta có thể chỉ định điều kiện đó.

Addition
The add() function sums the content of two arrays, and return the results in a new array
Phép cộng
Hàm add( )tính tổng nội dung của hai mảng và trả về kết quả trong một mảng mới..

In [1]:
# Add the values in arr1 to the values in arr2:
# Thêm các giá trị trong mảng1 vào các giá trị trong mảng2:
import numpy as np

arr1 = np.array([10, 11, 12, 13, 14, 15])
arr2 = np.array([20, 21, 22, 23, 24, 25])

newarr = np.add(arr1, arr2)

print(newarr)

[30 32 34 36 38 40]


The example above will return [30 32 34 36 38 40] which is the sums of 10+20, 11+21, 12+22 etc.
Ví dụ trên sẽ trả về [30 32 34 36 38 40] là tổng của 10+20, 11+21, 12+22, v.v.

In [ ]:
Subtraction
The subtract() function subtracts the values from one array with the values from another array, and return the results in a new array.
Phép trừ
Hàm subtract() trừ các giá trị từ một mảng bằng các giá trị từ mảng khác và trả về kết quả trong một mảng mới.

In [2]:
# Subtract the values in arr2 from the values in arr1:
# Trừ các giá trị trong mảng2 khỏi các giá trị trong mảng1:
import numpy as np

arr1 = np.array([10, 20, 30, 40, 50, 60])
arr2 = np.array([20, 21, 22, 23, 24, 25])

newarr = np.subtract(arr1, arr2)

print(newarr)

[-10  -1   8  17  26  35]


The example above will return [-10 -1 8 17 26 35] which is the result of 10-20, 20-21, 30-22 etc.
Ví dụ trên sẽ trả về [-10 -1 8 17 26 35] là kết quả của 10-20, 20-21, 30-22, v.v.

Multiplication
The multiply() function multiplies the values from one array with the values from another array, and return the results in a new array
Phép nhân
Hàm multiply( )nhân các giá trị từ một mảng với các giá trị từ một mảng khác và trả về kết quả trong một mảng mới..

In [3]:
# Multiply the values in arr1 with the values in arr2:
# Nhân các giá trị trong mảng1 với các giá trị trong mảng2:
import numpy as np

arr1 = np.array([10, 20, 30, 40, 50, 60])
arr2 = np.array([20, 21, 22, 23, 24, 25])

newarr = np.multiply(arr1, arr2)

print(newarr)

[ 200  420  660  920 1200 1500]


The example above will return [200 420 660 920 1200 1500] which is the result of 10*20, 20*21, 30*22 etc.
Ví dụ trên sẽ trả về [200 420 660 920 1200 1500] là kết quả của 10*20, 20*21, 30*22, v.v.

Division
The divide() function divides the values from one array with the values from another array, and return the results in a new array
Phép chia
Hàm divide( )chia các giá trị từ một mảng với các giá trị từ một mảng khác và trả về kết quả trong một mảng mới..

In [4]:
# Divide the values in arr1 with the values in arr2:
# Chia các giá trị trong mảng1 với các giá trị trong mảng2:
import numpy as np

arr1 = np.array([10, 20, 30, 40, 50, 60])
arr2 = np.array([3, 5, 10, 8, 2, 33])

newarr = np.divide(arr1, arr2)

print(newarr)

[ 3.33333333  4.          3.          5.         25.          1.81818182]


The example above will return [3.33333333 4. 3. 5. 25. 1.81818182] which is the result of 10/3, 20/5, 30/10 etc.
Ví dụ trên sẽ trả về [3.33333333 4. 3. 5. 25. 1.81818182] là kết quả của 10/3, 20/5, 30/10, v.v.

Power
The power() function rises the values from the first array to the power of the values of the second array, and return the results in a new array
Quyền lực
Hàm power( )tăng các giá trị từ mảng đầu tiên lên lũy thừa các giá trị của mảng thứ hai và trả về kết quả trong một mảng mới..

In [5]:
# Raise the valules in arr1 to the power of values in arr2:
# Nâng các giá trị trong mảng1 lên lũy thừa các giá trị trong mảng2:
import numpy as np

arr1 = np.array([10, 20, 30, 40, 50, 60])
arr2 = np.array([3, 5, 6, 8, 2, 33])

newarr = np.power(arr1, arr2)

print(newarr)

[      1000    3200000  729000000 -520093696       2500          0]


The example above will return [1000 3200000 729000000 6553600000000 2500 0] which is the result of 10*10*10, 20*20*20*20*20, 30*30*30*30*30*30 etc.
Ví dụ trên sẽ trả về [1000 3200000 729000000 6553600000000 2500 0] là kết quả của 10*10*10, 20*20*20*20*20, 30*30*30*30*30*30, v.v.

Remainder
Both the mod() and the remainder() functions return the remainder of the values in the first array corresponding to the values in the second array, and return the results in a new array
Còn lại
Cả hai hàm the mod()và remainder()hàm đều trả về phần còn lại của các giá trị trong mảng đầu tiên tương ứng với các giá trị trong mảng thứ hai và trả về kết quả trong một mảng mới..

In [6]:
# Return the remainders:
# Trả lại số dư:
import numpy as np

arr1 = np.array([10, 20, 30, 40, 50, 60])
arr2 = np.array([3, 7, 9, 8, 2, 33])

newarr = np.mod(arr1, arr2)

print(newarr)

[ 1  6  3  0  0 27]


The example above will return [1 6 3 0 0 27] which is the remainders when you divide 10 with 3 (10%3), 20 with 7 (20%7) 30 with 9 (30%9) etc.
Ví dụ trên sẽ trả về [1 6 3 0 0 27] là số dư khi bạn chia 10 với 3 (10%3), 20 với 7 (20%7) 30 với 9 (30%9), v.v.

You get the same result when using the remainder() function:
Bạn nhận được kết quả tương tự khi sử dụng remainder() hàm:

In [ ]:
# Return the remainders:
# Trả lại số dư:
import numpy as np

arr1 = np.array([10, 20, 30, 40, 50, 60])
arr2 = np.array([3, 7, 9, 8, 2, 33])

newarr = np.remainder(arr1, arr2)

print(newarr)

Quotient and Mod
The divmod() function return both the quotient and the the mod. The return value is two arrays, the first array contains the quotient and second array contains the mod
Thương số và Mod
Hàm divmod( )trả về cả thương và mod. Giá trị trả về là hai mảng, mảng đầu tiên chứa thương và mảng thứ hai chứa mod..

In [7]:
# Return the quotient and mod:
# Trả về thương và mod:
import numpy as np

arr1 = np.array([10, 20, 30, 40, 50, 60])
arr2 = np.array([3, 7, 9, 8, 2, 33])

newarr = np.divmod(arr1, arr2)

print(newarr)

(array([ 3,  2,  3,  5, 25,  1]), array([ 1,  6,  3,  0,  0, 27]))


The example above will return:
(array([3, 2, 3, 5, 25, 1]), array([1, 6, 3, 0, 0, 27]))
The first array represents the quotients, (the integer value when you divide 10 with 3, 20 with 7, 30 with 9 etc.
The second array represents the remainders of the same divisio
Ví dụ trên sẽ trả về:
(array([3, 2, 3, 5, 25, 1]), array([1, 6, 3, 0, 0, 27]))
Mảng đầu tiên biểu thị thương số, ( giá trị số nguyên khi bạn chia 10 cho 3, 20 cho 7, 30 cho 9, v.v.
Mảng thứ hai biểu thị số dư của cùng một phép chia.ns.

Absolute Values
Both the absolute() and the abs() functions do the same absolute operation element-wise but we should use absolute() to avoid confusion with python's inbuilt math.abs(
Giá trị tuyệt đối
Cả hàm absolute()và abs()hàm đều thực hiện cùng một phần tử hoạt động tuyệt đối nhưng chúng ta nên sử dụng absolute() để tránh nhầm lẫn với phần tử sẵn có của pythonmath.abs())

In [8]:
# Return the quotient and mod:
# Trả về thương và mod:
import numpy as np

arr = np.array([-1, -2, 1, 2, 3, -4])

newarr = np.absolute(arr)

print(newarr)

[1 2 1 2 3 4]


The example above will return [1 2 1 2 3 4].
Ví dụ trên sẽ trả về [1 2 1 2 3 4].